lib/ansible/module_utils/common/parameters.py

```diff
@@ -705,6 +705,7 @@ def _validate_sub_spec(
     no_log_values=None,
     unsupported_parameters=None,
     supported_parameters=None,
+    alias_deprecations=None,
```

Comment:
This function is recursive, so the new option needs to be added on line 824 too. Looks good to me besides that.

Only tangentially related, since you've looked at both, do you think `ArgumentSpecValidator.validate()`/`_validate_sub_spec()` should be consolidated? It seems like we're managing pretty much the same steps in two places.

Reply:
Thanks for spotting that! Should be fixed in c6654642c4b8fc65442d12fa3a1ae91a3b4c1ff3.



> Only tangentially related, since you've looked at both, do you think `ArgumentSpecValidator.validate()`/`_validate_sub_spec()` should be consolidated? It seems like we're managing pretty much the same steps in two places.

I think they definitely should! I was wondering already why the code exists twice...

Reply:
Ok, cool. I will probably attempt that after feature freeze.